### ISSUE TITLE
Possibly a DexMethod::set_deobfuscated_name issue?

### ISSUE THREAD

**Issue by fengruisd** [facebook/redex#786]
the input parameter `name` is normally generated by show(method), which means it is the full name of the method. on the other hand, the return value of `method->get_name()`, and the input paramter of `g_redex->erase_method` and `g_redex->get_method` is the simple name of a method.

So, would the place marked in the illustration below be a mistake?

![image](https://github.com/facebook/redex/assets/9607170/8fe006fc-17ca-4a10-ac35-fd5910161765)

**Comment by thezhangwei** [comment]
Yeah passing in the full name seems to be off. I think you are right on that.
However, `kInsertDeobfuscatedNameLinks` is defaulted to be false. So the deobfuscated name lookup is not being used. We found it to be problematic. Hope this helps. :)

**Comment by agampe** [comment]
On the other hand, thinking about it, though, the deobfuscated name may be with a different class name altogether? Just using the deobfuscated name wouldn't generally be enough to find the right method? Not sure how that ever worked correctly together at all.
